### PR TITLE
[keymgr/dv] Update testplan and covergroup plan

### PR DESCRIPTION
**Pull request by weicaiyang**
We can review this on V2 review meeting
Signed-off-by: Weicai Yang <weicai@google.com>

### CODE REVIEW

**Comment by weicaiyang** [comment]
Hi all, thanks for attending the keymgr V2 review. I have updated the testplan as discussed. Please take a look

File: hw/ip/keymgr/data/keymgr_testplan.hjson [hunk 1]
```diff
@@ -206,4 +209,82 @@
       tests: ["keymgr_stress_all"]
     }
   ]
+  covergroups: [
+    {
+      name: state_and_op_cg
+      desc: '''
+            - Cover all operations with cdi_sel, dest_sel and op_status (only fail or success)
```
Comment (cindychip) [inline hw/ip/keymgr/data/keymgr_testplan.hjson:216]: shall we use `` for CSR names? -> `cdi_sel`, `dest_sel` and `op_status`?

File: hw/ip/keymgr/data/keymgr_testplan.hjson [hunk 2]
```diff
@@ -206,4 +209,82 @@
...
+            any of all the states.
+            - Cover `sideload_clear` with any combination of availability of 3 sideload interfaces.
+            - This is sampled once sideload_clear is programmed after an operation.'''
+    }
+    {
+      name: reseed_interval_cg
+      desc: '''
+            - Cover small values of reseed_interval are used, so that TB can actually check EDN
+              request is sent in the right interval.
+            - Also Cover some large values to ensure all bits are toggled'''
```
Comment (cindychip) [inline hw/ip/keymgr/data/keymgr_testplan.hjson:240]: - Also c(lower case C)over some large values to ensure all bits are toggled. --> add a period?

File: hw/ip/keymgr/data/keymgr_testplan.hjson [hunk 3]
```diff
@@ -206,4 +209,82 @@
...
+    }
+    {
+      name: keymgr_sw_input_cg
+      desc: '''
+            - Cover all bits of SW inputs are toggled.
+            - SW input includes these CSRS: *_sw_binding, salt, key_version, max_*_key_ver*.
+            - Cross with the corresponding regwen'''
```
Comment (cindychip) [inline hw/ip/keymgr/data/keymgr_testplan.hjson:247]: period? and shall we add `` around CSRs in line 246?

File: hw/ip/keymgr/data/keymgr_testplan.hjson [hunk 4]
```diff
@@ -206,4 +209,82 @@
...
+    }
+    {
+      name: control_w_regwen_cg
+      desc: '''
+            - Similar to keymgr_sw_input_cg, create a separate cg as there are some reserved fields
+              scattered in the CSR
```
Comment (cindychip) [inline hw/ip/keymgr/data/keymgr_testplan.hjson:253]: period?

File: hw/ip/keymgr/data/keymgr_testplan.hjson [hunk 5]
```diff
@@ -206,4 +209,82 @@
...
+            - Cover `err_codes` values except `INVALID_SHADOW_UPDATE` as that is tested in a common
+              direct test.
+            - This is sampled when `err_codes` is read.'''
+    }
+    {
+      name: hw_invalid_input_cg
+      desc: '''
+            Cover all HW invalid inputs, including
+            - all ones/zeros on OTP root key.
+            - OTP root key valid is low
```
Comment (cindychip) [inline hw/ip/keymgr/data/keymgr_testplan.hjson:268]: missing a period? and line 271, 279?

**Review by cindychip — COMMENTED** [review]
Just a few small suggestions

**Review by cindychip — APPROVED** [review]

File: hw/ip/keymgr/data/keymgr_testplan.hjson [hunk 6]
```diff
@@ -54,33 +59,24 @@
...
     }
     {
       name: sideload
       desc: '''
-            Keymgr contains HW sideload interfaces to output keys for KMAC, HMAC, AES.
+            Keymgr contains HW sideload interfaces to output keys for KMAC, AES, OTBN.
 
             Stimulus:
-            Generate a keymgr output to HW sideload interface, excising all the sideload interfaces.
+            - Generate a keymgr output to HW sideload interface, excising all the sideload
```
Comment (matutem) [inline hw/ip/keymgr/data/keymgr_testplan.hjson:77]:
```suggestion
            - Generate a keymgr output to HW sideload interface, exercising all the sideload
```

**Review by matutem — APPROVED** [review]
LGTM with a suggestion.

File: hw/ip/keymgr/data/keymgr_testplan.hjson (hunk 6 above)
Comment (weicaiyang) [reply hw/ip/keymgr/data/keymgr_testplan.hjson]: Fix. Thanks, Matute.

**Pull request merged**